libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -258,14 +258,19 @@ struct _CCCL_TYPE_VISIBILITY_DEFAULT __type_defer
 {};
 
 // Implementation for indexing into a list of types:
-#  if defined(__cpp_pack_indexing)
+#  if defined(__cpp_pack_indexing) && defined(_CCCL_CUDA_COMPILER_CLANG)
```

Comment:
this changes it so that pack indexing is never used _except_ when using clang as a cuda compiler.

Reply:
No, NVC++ does not support pack indexing yet.

NVCC and NVC++ currently use the same version of the EDG front end.  EDG has not implemented pack indexing yet, and it is not know when they will implement it.

(But NVC++ won't define `__cpp_pack_indexing`, so the proposed change should actually work.  The problem with NVCC is that the host compiler controls the definition of `__cpp_pack_indexing` and other language-level feature test macros, even if NVCC doesn't support those features.  NVC++ doesn't have that problem.)